Pokemon Unity/Assets/Scripts2/Pokemon/Move.cs

```diff
@@ -8,2131 +8,2142 @@
-        private Move thismove { get; set; }	                                //= move
-		private int totalpp { get; set; }
-		/// <summary>
-		/// For Aerilate, Pixilate, Refrigerate
-		/// </summary>
-		private bool powerboost;		
+        public Move(Moves move, byte pp, int ppUp)
+        {
+            _base = new MoveData().getMove(move);
+            PP = pp;
+            PPups = ppUp;
+        }
+
+        #region Enumerator
+        public enum Effect
+        {
+            /// <summary>
+            /// Superclass that handles moves using a non-existent function code.
+            /// Damaging moves just do damage with no additional effect.
+            /// Non-damaging moves always fail.
+            /// </summary>
+            UnimplementedMove = -1,
+            /// <summary>
+            /// Superclass for a failed move. Always fails.
+            /// This class is unused.
+            /// </summary>
+            FailedMove = -2,
+            /// <summary>
+            /// Pseudomove for confusion damage.
+            /// </summary>
+            Confusion = -3,
+            /// <summary>
+            /// Implements the move Struggle.
+            /// For cases where the real move named Struggle is not defined.
+            /// </summary>
+            Struggle = -4,
+            //[Description("0x000")]
+            /// <summary>
+            /// No additional effect.
+            /// </summary>
+            x000 = 0x000,
+            /// <summary>
+            /// Does absolutely nothing. (Splash)
+            /// </summary>
+            x001 = 0x001,
+
+            /// <summary>
+            /// Struggle. Overrides the default Struggle effect above.
+            /// </summary>
+            x002 = 0x002,
+
+            /// <summary>
+            /// Puts the target to sleep.
+            /// </summary>
+            x003 = 0x003,
+
+            /// <summary>
+            /// Makes the target drowsy; it will fall asleep at the end of the next turn. (Yawn)
+            /// </summary>
+            x004 = 0x004,
+
+            /// <summary>
+            /// Poisons the target.
+            /// </summary>
+            x005 = 0x005,
+
+            /// <summary>
+            /// Badly poisons the target. (Poison Fang, Toxic)
+            /// (Handled in Pokemon's pbSuccessCheck): Hits semi-invulnerable targets if user
+            /// is Poison-type and move is status move.
+            /// </summary>
+            x006 = 0x006,
+
+            /// <summary>
+            /// Paralyzes the target.
+            /// Thunder Wave: Doesn't affect target if move's type has no effect on it.
+            /// Bolt Strike: Powers up the next Fusion Flare used this round.
+            /// </summary>
+            x007 = 0x007,
+
+            /// <summary>
+            /// Paralyzes the target. Accuracy perfect in rain, 50% in sunshine. (Thunder)
+            /// (Handled in Pokemon's pbSuccessCheck): Hits some semi-inv
```

Comment:
if you're going to store "TotalPP" into your Serialized code, store the value that writes into the PP (i think it was "PP-Ups Used" -- there's a limit to how many a pokemon can take). Dont save the end results, save the variables used in math equation. It's the variables that are important, not the equation/results (if math changes, pokemon data will still remain intact/uncorrupted).

Reply:
The SeriMove class doesn't take the total pp. It takes the Move, PP and PPups. PP in this case is the amount of PP that's left on the move before it was saved.